models/star.go

```diff
@@ -0,0 +1,89 @@
+package models
+
+// Star represents a starred repo by an user.
+type Star struct {
+	ID     int64 `xorm:"pk autoincr"`
+	UID    int64 `xorm:"UNIQUE(s)"`
+	RepoID int64 `xorm:"UNIQUE(s)"`
+}
+
+// StarRepo or unstar repository.
+func StarRepo(userID, repoID int64, star bool) error {
+	sess := x.NewSession()
+
```

Comment:
defer sess.Close()

This must be called and sess.Rollback will be invoked in Close if Commit will not be invoked.